crates/apollo-compiler/tests/validation/recursion.rs

```diff
@@ -105,6 +117,7 @@ fn long_fragment_chains_do_not_overflow_stack() {
     .expect_err("must have recursion errors");
 
     let expected = expect_test::expect![[r#"
+        Error: too much recursion
```

Comment:
This test now hits a recursion limit in two places, once in fragment validation and once in field merging validation, and emitting two errors for it is correct enough in my opinion